packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -0,0 +1,336 @@
+    <nimble-table-column-boolean-text
+        field-name="testResult"
+        true-message="Pass"
+        false-message="Fail"
+    >
+        Test Result
+    <nimble-table-boolean-text>
+</nimble-table>
+```
+
+##### Example E: Enum Text Column
+
+We may try to provide an easy way for clients to map enum values to localized strings. Here is a concept for an API which probably has issues.
+
+```html
+<nimble-table>
+    <nimble-table-column-enum-text
+        field-name="status"
+    >
+        Status
+        <nimble-list-option slot="enum-string-0" value="0">Pass</nimble-list-option>
+        <nimble-list-option slot="enum-string-1" value="1">Fail</nimble-list-option>
+    <nimble-table-column-enum-text>
+</nimble-table>
+```
+
+**Pros:**
+
+-   Easy for clients to use since configuration is declarative. This means Blazor apps could do configuration from .NET code and not need to write JS.
+-   Consistent numeric formatting across apps
+-   Easy to enforce styling differences between string and numeric columns (e.g. right vs left text alignment)
+
+**Cons:**
+
+-   Requires Nimble team to design simple but powerful formatting and i18n APIs
+-   Can't solve some use cases like app-specific formatting logic
+
+**Implementation Cost:**
+
+-   API design and implementation for each new column type
+
+#### Alternative 4: Client provides custom column implementation for each use case
+
+Nimble already has a mechanism for clients to provide custom columns by deriving from a base class, specifying the data fields / template / styling, and registering the column type with Nimble. We could ask clients to use this mechanism for text column types.
+
+**Pros:**
+
+-   Zero implementation cost to Nimble team
+-   Powerful; clients can format data however they want, including via browser APIs which are i18n-friendly
+
+**Cons:**
+
+-   Higher burden on clients to specify template, styling, numeric formatting, etc in JS. This is especially burdensome in frameworks like Blazor.
+-   Difficult to enforce styling differences between string and numeric columns (e.g. right vs left text alignment)
+-   Potential cross-app inconsistency if formatting code isn't shared
+
+### Strawman Proposal
+
+For the sake of discussion my initial proposal is:
+
+1. We need to offer an approach for columns that require app-specific formatting logic to support above use cases like 5 (custom unit logic) and 8 (data of unknown type).
+    - I believe the cons of **Client provides a custom column implementation for each use case** are too great so we should invest in an approach that offers clients more consistency and simplicity.
+    - I'm leaning towards **Client specifies formatting function** over **Use `table-column-text`** because it more clearly encodes that the column is numeric, giving better type safety and allowing for more consistent styling. I'd like to do performance profiling to see how it impacts scroll performance before committing to this direction.
```

Comment:
Some related statements to discuss:

1. If a client needs to write any JS then implementing a column plugin isn't a big leap from there anyway? Maybe even having the overhead will encourage contributing it to nimble instead of bespoke stuff being everywhere?
2. If you are defining a custom formatter as a client you're one tiny nudge away from making it parameterized. Encourage parameterizing via a declarative custom element API instead of clients creating formatter function factories.
3. I don't think I want clients to ever be exposed to setting callbacks on our custom elements as publicly supported APIs. Plugin authors overriding a base class, overriding a formatter method, and defining components are fine.

So I guess the proposal diff would be:
1. Create an easy to extend FormatedTextColumn base class that very few folks outside nimble need (use case 5, and I think use case 8 doesn't actually apply to this spec).
2. Nimble provides column implementation for common use cases. They are implemented by extending the FormattedTextColumn base class.

Reply:
It's not clear to me what we are going to consider "common use cases" that we will plan on having within nimble. Would that include the cases mentioned in use case 5 of the elapsed time column and the file size column?

One current obstacle of adding formatted columns in nimble (whether by us out-of-the-box or encouraged by clients) is that localization isn't straightforward with nimble. For something like an elapsed time column or a file size column I guess we'd have attributes (or some other way) to specify each localized unit such as `days-unit`, `hours-unit`, `kilobyte-unit`, or `byte-unit`?

Reply:
I believe the update in #1224 captures the direction in Milan's comment. It still doesn't go far enough to hit the localization cases that Molly mentioned. For now those could either be client-specified columns or we could design them alongside a solution to #1090.